sqs/sqs.go

```diff
@@ -3,80 +3,96 @@ package sqs
-	DeduplicationScope            string `json:"DeduplicationScope,omitempty"`
-	FifoThroughputLimit           string `json:"FifoThroughputLimit,omitempty"`
+type QueueConfigAttributes struct {
+	DelaySeconds                  string                    `json:"DelaySeconds,omitempty"`
+	MaximumMessageSize            string                    `json:"MaximumMessageSize,omitempty"`
+	MessageRetentionPeriod        string                    `json:"MessageRetentionPeriod,omitempty"`
+	Policy                        string                    `json:"Policy,omitempty"`
+	ReceiveMessageWaitTimeSeconds string                    `json:"ReceiveMessageWaitTimeSeconds,omitempty"`
+	RedrivePolicy                 string                    `json:"RedrivePolicy,omitempty"`
+	DeadLetterTargetArn           string                    `json:"deadLetterTargetArn,omitempty"`
+	MaxReceiveCount               string                    `json:"maxReceiveCount,omitempty"`
+	VisibilityTimeout             string                    `json:"VisibilityTimeout,omitempty"`
+	KmsMasterKeyId                string                    `json:"KmsMasterKeyId,omitempty"`
+	KmsDataKeyReusePeriodSeconds  string                    `json:"KmsDataKeyReusePeriodSeconds,omitempty"`
+	SqsManagedSseEnabled          string                    `json:"SqsManagedSseEnabled,omitempty"`
+	FifoQueue                     QueueConfigAttributesBool `json:"FifoQueue,omitempty"`
+	ContentBasedDeduplication     QueueConfigAttributesBool `json:"ContentBasedDeduplication,omitempty"`
+	DeduplicationScope            string                    `json:"DeduplicationScope,omitempty"`
+	FifoThroughputLimit           string                    `json:"FifoThroughputLimit,omitempty"`
 }
 
-func (q QueueConfigAtrributes) Attributes() map[string]string {
-	b, _ := json.Marshal(q)
+// QueueConfigAttributesBool is a custom type for bool values in QueueConfigAttributes
+// that supports marshaling to string.
+type QueueConfigAttributesBool bool
+
+func (q QueueConfigAttributesBool) MarshalText() ([]byte, error) {
+	if q {
+		return []byte("true"), nil
+	}
+	return []byte("false"), nil
+}
+
+func (q QueueConfigAttributes) Attributes() (map[string]string, error) {
+	b, err := json.Marshal(q)
+	if err != nil {
+		return nil, fmt.Errorf("cannot marshal queue attributes (json.Marshal): %w", err)
+	}
+
 	var m map[string]string
-	_ = json.Unmarshal(b, &m)
-	return m
+	err = json.Unmarshal(b, &m)
+	if err != nil {
+		return nil, fmt.Errorf("cannot marshal queue attributes (json.Unmarshal): %w", err)
+	}
+
+	return m, nil
 }
 
-func GetQueueUrl(ctx context.Context, sqsClient *sqs.Client, topic string) (*string, error) {
-	getQueueOutput, err := sqsClient.GetQueueUrl(ctx, &sqs.GetQueueUrlInput{
-		QueueName: aws.String(topic),
-	})
+func getQueueUrl(ctx context.Context, sqsClient *sqs.Client, topic string, generateGetQueueUrlInput GenerateGetQueueUrlInputFunc) (*string, error) {
```

Comment:
I made it private to not increase public API surface (and now it requires generateGetQueueUrlInput), but if anybody see it useful please let me know